### PR TITLE
Use replicas field when scheduling pods

### PR DESCRIPTION
**Pull request by maltesander**
The replicas field in the role group now is taken into account when scheduling pods or deleting excess pods. 
For now it is simply based on a random number (which nodes are selected if multiple match or which pods are taken of nodes if excess).

### CODE REVIEW

File: src/role_utils.rs [hunk 1]
```diff
@@ -189,19 +189,22 @@ where
...
+            group_name.clone(),
+            (nodes, usize::from(role_group.replicas)),
+        );
     }
     Ok(found_nodes)
 }
 
-/// Return a list of eligible nodes for each role and group combination.
-/// Required to delete excess pods that do not match any node or selector description.
+/// Return a list of eligible nodes and the provided replicas count for each role and group
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply src/role_utils.rs:200]: Also: This could now benefit from a proper comment which explains the input type.

**Review by lfrancke — CHANGES_REQUESTED** [review]
Somehow I can't believe that this is it.....my only comments are about documentation I believe.

I have a nagging feeling that we're missing something though.

File: src/k8s_utils.rs [hunk 2]
```diff
@@ -94,52 +102,122 @@ pub fn find_nodes_that_need_pods<'a>(
     candidate_nodes: &'a [Node],
     existing_pods: &[Pod],
     label_values: &BTreeMap<String, Option<String>>,
+    replicas: usize,
 ) -> Vec<&'a Node> {
-    candidate_nodes
+    let nodes_that_need_pods = candidate_nodes
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/k8s_utils.rs:107]: But that is still true for one role group? I updated the comment with a hint on that.

File: src/k8s_utils.rs [hunk 3]
```diff
@@ -97,7 +100,12 @@ pub fn find_valid_pods_for_nodes<'a>(
 /// NameNode Pod.
 /// And this is the label you can now filter on using the `label_values` argument.
 ///
+/// Additionally the replicas field of a role group is taken into account. When selecting nodes,
+/// a random subset representing the size difference of "replicas" and "nodes_that_need_pods" is selected.
```
Comment (lfrancke) [inline src/k8s_utils.rs:104]:
```suggestion
/// a random subset representing the size difference between "replicas" and "nodes_that_need_pods" is selected.
```

**Review by lfrancke — DISMISSED** [review]
Minor comment things

**Review by soenkeliebau — APPROVED** [review]
As Lars is on vacation I've gone over his comments and that all looks good to me.

**Pull request merged**